js-packages/@prestojs/util/src/pagination/CursorPaginator.ts

```diff
@@ -199,4 +200,36 @@ export default class CursorPaginator extends Paginator<
     hasNextPage(): boolean {
         return !!this.internalState.nextCursor;
     }
+
+    static getPaginationState(
+        requestDetails: PaginationRequestDetails
+    ): Record<string, any> | false {
+        const { decodedBody } = requestDetails;
+        // If it's an array then it's assumed to be unpaginated data
+        if (Array.isArray(decodedBody) || !decodedBody) {
+            return false;
+        }
+
+        // Cursor pagination responses should contain a next and previous link that includes
+        // a query parameter with name 'cursor' containing the next/previous cursor value. Results
+        // should be an array under the `results` key.
+        const isCursorLink = (link: any): boolean =>
+            typeof link == 'string' && 'cursor' in qs.parse(qs.extract(link));
+        if (isCursorLink(decodedBody.next) || isCursorLink(decodedBody.previous)) {
```

Comment:
what happens if both next & previous are null? (ie. paginated but single page)

Reply:
yes good catch - existing code was buggy in this way

i've pushed a fix now